src/common/Edelstein.Common.Gameplay.Game/Combat/AttackRequest.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Edelstein.Common.Utilities.Packets;
+using Edelstein.Protocol.Gameplay.Game.Combat;
+using Edelstein.Protocol.Utilities.Packets;
+
+namespace Edelstein.Common.Gameplay.Game.Combat;
+
+public class AttackRequest : IAttackRequest, IPacketReadable
+{
+    public AttackRequest(AttackType type) => Type = type;
+
+    public AttackType Type { get; }
+    
+    public int DamagePerMob { get; private set; }
+    public int MobCount { get; private set; }
+    
+    public int SkillID { get; private set; }
+    public int Keydown { get; private set; }
```

Comment:
In C++ or C# envinronment and in context of system events KeyDown, KeyUp i would prefer naming it KeyDown

In web env like using js i would use Keydown.

Reply:
I'll adjust this when I work on skills that use KeyDown